### PR TITLE
Fix index flag doesn't work in SQL #11035

### PR DESCRIPTION
**Pull request by Oceania2018**

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Indexing/SQL/TextFieldIndexProvider.cs [hunk 1]
```diff
@@ -91,6 +92,10 @@ public override void Describe(DescribeContext<ContentItem> context)
                             continue;
                         }
 
+                        var contentIndexSettings = fieldDefinition.GetSettings<ContentIndexSettings>();
+                        if (!contentIndexSettings.Included)
+                            continue;
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.ContentFields/Indexing/SQL/TextFieldIndexProvider.cs:97]:
```suggestion
                        if (!contentIndexSettings.Included)
                        {
                            continue;
                        }                            
```

**Comment by sebastienros** [comment]
Also I believe this change would have to be done on all index types, not just for TextFields.

**Comment by ns8482e** [comment]
As currently it index all regardless of the flag - What would be the impact on previously indexed field ?  Will it remove those index if it's unchecked?

**Comment by Oceania2018** [comment]
> What would be the impact on previously indexed field

It will remove the index if uncheck the flag.
It prevent some unnecessary data index data growing too fast.

**Comment by ns8482e** [comment]
@Oceania2018 Isn't this a breaking change -  as it will not index currently indexed field 

IMO for YSQL index - it should have new property like `SQLIncluded` with default to `true` than reusing  `Included` to avoid breaking changes.

**Comment by deanmarcussen** [comment]
@Skrypt I think we should probably have a different setting class / driver etc for sql field indexes, as you are already doing on your other pr, for elastic. 

The value could default to `true`, to prevent breaking changes as per @ns8482e comment.

Thoughts?

**Comment by Oceania2018** [comment]
@sebastienros What do you think? I'm relatively new to OrchardCore.

**Comment by Skrypt** [comment]
@deanmarcussen Yes, we need a separate setting for SQL index field settings. We should not try to reuse the Lucene Index ones.

**Comment by Oceania2018** [comment]
> @Oceania2018 Isn't this a breaking change - as it will not index currently indexed field after this change?
> 
> IMO for YSQL index - it should have new property like `SQLIncluded` with default to `true` than reusing `Included` to avoid breaking changes.

When I set default as `True`, lt always show as `True` even I unchecked it and saved.
```csharp
public bool SQLIncluded { get; set; } = true;
```
Do you know how to set default in the correct way?

**Comment by sebastienros** [comment]
@Oceania2018 yes it needs custom settings then.

I don't think the default value should be indexed, I'd rather see a migration step that does it just for backward compatibility, but a newly enabled feature should be opt-in.

We should also move the current indexing settings to a custom Lucene one, with migration too, in a separate PR (not sure it's done in ES).

**Comment by Oceania2018** [comment]
@sebastienros 
Already added the custom settings, but keep the value as true by default in order to not breaking other things.
Do you know when this can be merged?

**Comment by Skrypt** [comment]
This is not an appropriate fix as I said before in another issue. Here, instead of adding a `SQLIncluded` field on that class, we need to abstract the implementation with an Interface as I did in this PR: #10515.

I suggest we wait and finish #10515 before getting back on this one. That's a fairly easy PR to adjust afterward.

**Comment by Piedone** [comment]
Closing due to inactivity.

**Pull request closed without merging**